### PR TITLE
Config updates

### PR DESCRIPTION
**Pull request by adrienbernede**
## Spack configs
- [X] Corona: Add ROCm 6.0.2 -> same error as with tioga, need ROCm 6.1.x , reverted to ROCm 5.7.1
- [X] Tioga: Add ROCm 6.1.2 and 6.2.0. Add cce 18.0.0
- [X] Add latest toolchains on lassen

## Shared Jobs
- [X] Test rocm ~~6.1.2~~ 6.2.0 on tioga
- [X] Remove Intel 19
- [X] Update to Intel 2023
- [X] Remove CUDA 10 job (blueos system default is now 11.2.0)
- [X] Test latest toolchains on lassen
- [X] Remove XL jobs, not used in production, but keep the toolchain available 
- [X] Add cce 18.0.0, although this spec is causing issues in RAJA and Caliper

## Testing
- [ ] https://github.com/LLNL/CHAI/pull/275
- [X] https://github.com/LLNL/RAJA/pull/1712
- [X] https://github.com/LLNL/Umpire/pull/902
- [X] https://github.com/LLNL/CARE/pull/281
- [ ] https://github.com/LLNL/Caliper/pull/588

### CODE REVIEW

File: blueos_3_ppc64le_ib/spack.yaml [hunk 1]
```diff
@@ -698,6 +724,8 @@ spack:
         prefix: /usr/tce/packages/spectrum-mpi/spectrum-mpi-rolling-release-clang-16.0.6
       - spec: spectrum-mpi@10.3.1.03rtm0%clang@=16.0.6.gcc.8.3.1
         prefix: /usr/tce/packages/spectrum-mpi/spectrum-mpi-rolling-release-clang-16.0.6-gcc-8.3.1
+      - spec: spectrum-mpi@10.3.1.03rtm0%clang@=16.0.6.cuda.11.8.0.gcc.11.2.1.cuda.11.8.0
```
Comment (adayton1) [inline blueos_3_ppc64le_ib/spack.yaml:727]: cuda.11.8.0 is in the spec twice. Is that intentional?

File: gitlab/radiuss-jobs/lassen.yml [hunk 2]
```diff
@@ -23,15 +23,38 @@ clang_14_0_5_gcc_8_3_1_cuda_11_7_0:
...
     MODULE_LIST: "cuda/11.2.0"
   extends: .job_on_lassen
 
+xl_2023_06_28_gcc_11_2_1_cuda_11_8_0:
+  variables:
+    SPEC: "${PROJECT_LASSEN_VARIANTS} +cuda %xl@=16.1.1.14.cuda.11.8.0.gcc.11.2.1 ^cuda@11.8.0+allow-unsupported-compilers ${PROJECT_LASSEN_DEPS}"
+    MODULE_LIST: "cuda/11.8.0"
+  extends: .job_on_lassen
+
 gcc_8_3_1_cuda_11_7_0:
```
Comment (adayton1) [inline gitlab/radiuss-jobs/lassen.yml:50]: Do we need the CUDA 11.7 builds? I think most teams are on 11.2 or 11.8.

Comment (rhornung67) [reply gitlab/radiuss-jobs/lassen.yml:50]: Good catch. 11.2 is the default and what most teams are using. Some projects also test and work with 11.8. So let's go with 11.2 and 11.8 and nix 11.7

File: blueos_3_ppc64le_ib/spack.yaml (hunk 1 above)
Comment (adrienbernede) [reply blueos_3_ppc64le_ib/spack.yaml:727]: Oops

Comment (adrienbernede) [reply blueos_3_ppc64le_ib/spack.yaml:727]: It’s me trying to choose between LC notations and ours.

File: gitlab/radiuss-jobs/lassen.yml (hunk 2 above)
Comment (adrienbernede) [reply gitlab/radiuss-jobs/lassen.yml:50]: OK, but 11.7 is often associated with gcc 11.2.1, we already have gcc 11.2.1 + cuda 11.8.
I think this particular spec was there to test with the newest cuda. I think we can remove it in favor of gcc 11.2.1 + cuda 11.8.

Comment (rhornung67) [reply gitlab/radiuss-jobs/lassen.yml:50]: works for me.

Comment (adrienbernede) [reply gitlab/radiuss-jobs/lassen.yml]: Done.

**Comment by adrienbernede** [comment]
@rhornung67 @adayton1 I’d like an approval to merge this. You’ll find proof that this is working in the PRs mentioned above.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**